knowledge_storm/rm.py

```diff
@@ -539,3 +540,71 @@ def forward(self, query_or_queries: Union[str, List[str]], exclude_urls: List[st
                 continue
 
         return collected_results
+
+
+class SearXNG(dspy.Retrieve):
+    def __init__(self, searxng_api_url, searxng_api_key=None, k=3, is_valid_source: Callable = None):
+        """
+        Initialize the SearXNG search retriever.
+
+        Args:
+            searxng_api_url (str): The URL of the SearXNG API.
```

Comment:
searxng_api_url depends on your installation of SearXNG, usually it is on localhost:8080. I run SearXNG in docker, with api access configured to be on localhost:4000. searxng_api_key is None. In case of doubt please consult SearXNG [documentation](https://docs.searxng.org/index.html).